portal/models/user.py

```diff
@@ -1671,8 +1670,13 @@ def check_role(self, permission, other_id):
 
         abort(401, "Inadequate role for {} of {}".format(permission, other_id))
 
-    def has_role(self, role_name):
-        return role_name in [r.name for r in self.roles]
+    def has_role(self, role_name_or_list):
+        """Given role (or list of) returns true if user has at least one"""
+        if isinstance(role_name_or_list, str):
+            return role_name_or_list in [r.name for r in self.roles]
+        for item in role_name_or_list:
+            if item in [r.name for r in self.roles]:
```

Comment:
might as well define that inline list as a variable, defined at the beginning of the method; easier to read and doesn't re-create a list of the user's roles for every given role

Reply:
well, for those of us who love list comprehension, i'm not sure it's more legible, but i've plugged in a set in the name of "efficiency" - hard to imagine it'll matter with such a small 'n'